File: LeetCode Problems/1 to 100/56. Merge Intervals/56. Merge Intervals.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;
import java.util.List;

class Solution {
    public int[][] merge(int[][] intervals) {
        int n = intervals.length;
        Arrays.sort(intervals, Comparator.comparingInt(i -> i[0]));
        List<int[]> list = new ArrayList<>();
        if (n == 1) {
            return intervals;
        }
        int[] interval = intervals[0];
        list.add(interval);
        for (int[] interv : intervals) {
            if (interv[0] <= interval[1]) {
                interval[1] = Math.max(interval[1], interv[1]);

            } else {
                interval = interv;
                list.add(interval);
            }
        }
        return list.toArray(new int[list.size()][]);

    }
}

File: LeetCode Problems/1 to 100/56. Merge Intervals/56. Merge Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>> &ar) {
        sort(ar.begin(), ar.end());
        vector<vector<int>> ans;
        for (int i = 0; i < ar.size(); i++) {
            if (ans.size() == 0 or ans.back()[1] < ar[i][0]) {
                ans.push_back({ar[i][0], ar[i][1]});
            } else
                ans.back()[1] = max(ans.back()[1], ar[i][1]);
        }
        return ans;
    }
};